### PR TITLE
Homework 4 is done

### PR DESCRIPTION
**Pull request by chartal**

### CODE REVIEW

File: app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.android_academy.chartal_application.R
+import com.android_academy.chartal_application.data.Actor
+
+
+class ActorAdapter() :
+    RecyclerView.Adapter<ActorAdapter.ActorViewHolder>() {
+
+    val items = mutableListOf<Actor>()
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt]: in better to do it private field

File: app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    fun addItems(newItems: List<Actor>) {
+        items.addAll(newItems)
+        notifyDataSetChanged()
+    }
+
+    inner class ActorViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+        fun bind(actor: Actor) {
+            itemView.findViewById<ImageView>(R.id.iv_actor).setImageResource(actor.image)
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt]: for every onBind call findViewId not good
let's use kotlinExtension with layoutContainer
or do it inside init{}

File: app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    override fun onBindViewHolder(holder: ActorViewHolder, position: Int) {
+        holder.bind(items[position])
+    }
+
+    fun addItems(newItems: List<Actor>) {
+        items.addAll(newItems)
+        notifyDataSetChanged()
+    }
+
+    inner class ActorViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt]: inner keyword not need here

File: app/src/main/java/com/android_academy/chartal_application/adapters/MovieAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    interface Listener {
+        fun oItemClicked(movie: Movie)
+    }
+
+    inner class MovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+        fun bind(movie: Movie) {
+            itemView.setOnClickListener {
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/adapters/MovieAdapter.kt:53]: if you create onClickListener on onBind you create every time when you scroll your view
better create it on init

File: app/src/main/java/com/android_academy/chartal_application/data/Movie.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+@Parcelize
+data class Movie(
+    val title: String,
+    val description: String,
+    val review: String,
+    val overview: String,
+    val age: String,
+    var rating: Float,
+    val listActors: List<Actor>,
+    @DrawableRes val posterRes: Int,
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/data/Movie.kt:17]: here also same sitution

File: app/src/main/java/com/android_academy/chartal_application/data/Actor.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.android_academy.chartal_application.data
+
+import android.os.Parcelable
+import androidx.annotation.DrawableRes
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
+data class Actor(
+    @DrawableRes val image: Int,
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/data/Actor.kt:9]: it's not required but better be if you download it via url

File: app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesDetails.kt [hunk 7]
```diff
@@ -5,16 +5,20 @@ import android.os.Bundle
...
+import com.android_academy.chartal_application.data.Actor
+import com.android_academy.chartal_application.data.Movie
 import com.android_academy.chartal_application.databinding.FragmentMovieDetailsBinding
 
 class FragmentMoviesDetails : Fragment(R.layout.fragment_movie_details) {
 
     private var _binding: FragmentMovieDetailsBinding? = null
     private val binding get() = _binding!!
     private var listener: TransactionsFragmentClicks? = null
+    private val actorAdapter = ActorAdapter()
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesDetails.kt]: would be better in such cases use [`lazy`](https://medium.com/mobile-app-development-publication/kotlin-variable-to-be-lazy-or-to-be-late-ab865c682d61) patter

File: app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesList.kt [hunk 8]
```diff
@@ -6,14 +6,18 @@ import android.view.LayoutInflater
...
+import com.android_academy.chartal_application.repository.DataStorage
 
 
-class FragmentMoviesList : Fragment() {
+class FragmentMoviesList : Fragment(), MovieAdapter.Listener {
 
     private var _binding: FragmentMoviesListBinding? = null
     private val binding get() = _binding!!
     private var listener: TransactionsFragmentClicks? = null
+    private val movieAdapter = MovieAdapter(this)
```
Comment (Axrorxoja) [inline app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesList.kt]: here also the same situation will be good if you use lazy pattern but not required

File: app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt (hunk 1 above)
Comment (chartal) [reply app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt]: Внесено исправление: Свойство items сделано приватным

File: app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt (hunk 2 above)
Comment (chartal) [reply app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt]: Исправлено для ActorViewHolder и MovieViewHolder: созданы глобальные переменные

File: app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt (hunk 3 above)
Comment (chartal) [reply app/src/main/java/com/android_academy/chartal_application/adapters/ActorAdapter.kt]: исправлено: убрал модификатор inner

File: app/src/main/java/com/android_academy/chartal_application/adapters/MovieAdapter.kt (hunk 4 above)
Comment (chartal) [reply app/src/main/java/com/android_academy/chartal_application/adapters/MovieAdapter.kt:53]: исправлено:  перенесено в блок init{}

File: app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesDetails.kt (hunk 7 above)
Comment (chartal) [reply app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesDetails.kt]: исправлено:  использован lazy patter

File: app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesList.kt (hunk 8 above)
Comment (chartal) [reply app/src/main/java/com/android_academy/chartal_application/details/FragmentMoviesList.kt]: исправлено:  использован lazy patter

**Review by Axrorxoja — APPROVED** [review]

**Pull request merged**